### ISSUE TITLE
How are the multi-outputs affect the loss during training?

### ISSUE THREAD

**Issue by PJY-609** [raun1/ISBI_2018_LUNG_CLASSIFICATION#2]
In the paper, there are several *intermediate outputs* from different layers. The model is defined to have multi-outputs [code](https://github.com/raun1/ISBI2018-Diagnostic-Classification-Of-Lung-Nodules-Using-3D-Neural-Networks/blob/853fc93fe8ed19ce9c588e3d13de48606243f1fe/src/Isbi_2018_Lung_Nodule_Classification.py#L736).  
But I can't figure out how those *intermediate outputs* affect tranining, since you only define one `bianary_crossentropy` compiling the model. So how are those *intermediate outputs* affecting your training? Thanks?

**Comment by raun1** [comment]
Hello,

The loss is same for all the nodes. Since there is no change in the loss the same binary cross entropy is automatically applied to all the output layers.

Thank you

**Comment by PJY-609** [comment]
Thank you for your reply!

Are all the losses of those outputs equally weighted and summed up as the training loss?

**Comment by raun1** [comment]
Hello

Yes you are correct :).
Thank you

**Comment by PJY-609** [comment]
Good to know that 

Thanks !
